src/shapes/image.class.ts

```diff
@@ -1,868 +1,832 @@
+export class Image extends FabricObject {
+  /**
+   * When calling {@link getSrc}, return value from element src with `element.getAttribute('src')`.
+   * This allows for relative urls as image src.
+   * @since 2.7.0
+   * @type Boolean
+   * @default
+   */
+  srcFromAttribute: boolean;
+
+  /**
+   * private
+   * contains last value of scaleX to detect
+   * if the Image got resized after the last Render
+   * @type Number
+   */
+  protected _lastScaleX: number;
+
+  /**
+   * private
+   * contains last value of scaleY to detect
+   * if the Image got resized after the last Render
+   * @type Number
+   */
+  protected _lastScaleY: number;
+
+  /**
+   * private
+   * contains last value of scaling applied by the apply filter chain
+   * @type Number
+   */
+  protected _filterScalingX: number;
+
+  /**
+   * private
+   * contains last value of scaling applied by the apply filter chain
+   * @type Number
+   */
+  protected _filterScalingY: number;
+
+  /**
+   * minimum scale factor under which any resizeFilter is triggered to resize the image
+   * 0 will disable the automatic resize. 1 will trigger automatically always.
+   * number bigger than 1 are not implemented yet.
+   * @type Number
+   */
+  minimumScaleTrigger: number;
+
+  /**
+   * key used to retrieve the texture representing this image
+   * @since 2.0.0
+   * @type String
+   * @default
+   */
+  cacheKey: string;
+
+  /**
+   * Image crop in pixels from original image size.
+   * @since 2.0.0
+   * @type Number
+   * @default
+   */
+  cropX: number;
+
+  /**
+   * Image crop in pixels from original image size.
+   * @since 2.0.0
+   * @type Number
+   * @default
+   */
+  cropY: number;
+
+  /**
+   * Indicates whether this canvas will use image smoothing when painting this image.
+   * Also influence if the cacheCanvas for this image uses imageSmoothing
+   * @since 4.0.0-beta.11
+   * @type Boolean
+   * @default
+   */
+  imageSmoothing: boolean;
+
+  protected src: string;
+
+  preserveAspectRatio: string;
+
+  static filters: any;
+  filters: any[];
+  resizeFilter: any;
+
+  protected _element: ImageSource;
+  protected _originalElement: ImageSource;
+  protected _filteredEl: ImageSource;
+
+  /**
+   * Constructor
+   * Image can be initialized with any canvas drawable or a string.
+   * The string should be a url and will be loaded as an image.
+   * Canvas and Image element work out of the box, while videos require extra code to work.
+   * Please check video element events for seeking.
+   * @param {HTMLImageElement | HTMLCanvasElement | HTMLVideoElement | String} element Image element
+   * @param {Object} [options] Options object
+   */
+  constructor(element: ImageSource, options: any = {}) {
+    super(options);
+    this.filters = [];
+    this.cacheKey = `texture${FabricObject.__uid++}`;
+    this._initElement(element, options);
+  }
+
+  /**
+   * Returns image element which this instance if based on
+   */
+  getElement() {
+    return this._element;
```

Comment:
changed

Reply:
this is fine, returning the empty object wasn't really smart.